### DISCUSSION TITLE
Am I importing Tailwind correctly? Or is this a bug?

### DISCUSSION THREAD

**Discussion by evanb2** [crxjs/chrome-extension-tools#238] [Q&A] [answered]
Checkout this example repo: https://github.com/evanb2/extend-chrome-bug-repro

I'm noticing my content-script CSS is somehow bleeding out/clashing with the external page.

The only reason I'm not sure if this is something of my own doing or a bug with this plugin is that it only happens for production builds. But my first thought was that this is because Tailwind is probably doing a bunch of resets on HTML elements.

If you run `npm run dev` and navigate to Google Calendar, everything looks fine. 
If you run `npm run build` and refresh, you'll notice a bunch of layout issues in Google Calendar (see screenshot).

`Development build:`
<img width="863" alt="Screen Shot 2022-03-14 at 2 09 00 PM" src="https://user-images.githubusercontent.com/11276439/158261840-6b5ff4a5-8271-42ad-bbd8-4409781ebf8f.png">

`Production build:`
<img width="863" alt="Screen Shot 2022-03-14 at 2 04 52 PM" src="https://user-images.githubusercontent.com/11276439/158261604-a563e6cf-6b6e-4e5c-8ad5-28e6bcc113e8.png">

Thanks for any help!

**Comment by evanb2** [comment]
Ok this seems to be an issue with the Tailwind base reset ("preflight") not being scoped to the extension. Still curious why it doesn't happen in `development` mode. Also curious if anyone else out there has found a solution for this...

**Comment by jacksteamdev** [comment, marked as answer]
Tailwind setup LGTM. When RPCE builds the content script, it takes imported CSS assets and adds it to the manifest, which causes the scoping problem. There should be a way around this, I'll look into it.

**Reply by jacksteamdev** [reply]
Looking at your repo, I think you've got the right way to do styles injection for content scripts. I'll probably remove adding styles to the manifest.

**Reply by evanb2** [reply]
Oh interesting, thanks for that context. You're right, removing the `css` file from the manifest fixes everything.

Which is awesome cause otherwise it would have been a massive PITA haha.

**Reply by jacksteamdev** [reply]
I've explored this today, and it's trickier than I thought.

### Vite & Imported CSS

`vite build` concatenates styles into a single stylesheet, but the value of `import styles from './styles.css'` seems to include only the styles from `styles.css`, not the styles from imported SFCs. This is probably not our intention, since styles in Vue components would be lost.

What we want is a way to import the stylesheet URL, but Vite doesn't seem to support this. See https://github.com/vitejs/vite/issues/2522 A PR is in the works for `*.css?raw` imports, but I don't think it will help our use case: https://github.com/vitejs/vite/pull/5796

I could probably implement a virtual module that will export the path to the build output file, but that file doesn't exist in development. :thinking:

I'll admit I'm not a Vue expert. The Vue docs focus heavily on Custom Elements, which use Shadow DOM, but are not the same (and require polyfill in content scripts).

**Reply by jacksteamdev** [reply]
> removing the `css` file from the manifest fixes everything.

@evanb2 Happy to hear, I could make disabling this an option. Do the styles in your SFCs render correctly?

**Reply by evanb2** [reply]
Lol it's always something.

Just tested and SFC styles do _not_ work. Not a huge deal for me personally cause I haven't been using them. But yeah definitely something you'll want to support for Vue users.

**Reply by evanb2** [reply]
> I could make disabling this an option

I think that'd be the way to go.

**Reply by evanb2** [reply]
So this is a prod build, and the SFC styles file is there but I'm not seeing it work in the browser. Not sure if I'm missing something...

The `main.js.css` is also correctly referenced in the generated `manifest.json`.

![Screen Shot 2022-03-15 at 11 42 49 AM](https://user-images.githubusercontent.com/11276439/158449193-ef415e3b-30d9-497d-87be-f627ffa464b2.png)

**Reply by evanb2** [reply]
Ah but I'm still adding the styles manually to the shadow DOM, that's why.

Sorry for so many replies haha.

**Reply by jacksteamdev** [reply]
> The Vue docs focus heavily on Custom Elements, which use Shadow DOM, but are not the same (and require polyfill in content scripts).

BTW, content scripts don't need the full Web Components polyfill, just this library: [Custom Elements (v1) Polyfill](https://github.com/webcomponents/polyfills/tree/master/packages/custom-elements). See this Chromium bug for context: [390807 - Content scripts can't define custom elements - chromium](https://bugs.chromium.org/p/chromium/issues/detail?id=390807)

#224 adds this package during development to support the Vite Error Overlay (which uses custom elements). I'm considering auto-including it during production if a content script calls `customElements.define`. It's probably an unnecessary distraction for the developer.
